std/container/array.d

```diff
@@ -349,7 +356,7 @@ if (!is(Unqual!T == bool))
             import core.checkedint : mulu;
             bool overflow;
             const sz = mulu(elements, T.sizeof, overflow);
-            if (overflow) assert(0);
+            assert(!overflow, "Trying to allocate more than size_t.max bytes.");
```

Comment:
Thank you for the explanation. But honestly, for me the sentence:
>  The optimization and code generation phases of compilation may assume that it is unreachable code.

means that it *may* be removed in release mode. For example, on my computer below code stops with `SIGSEV` on release mode, for both the old and mine version of `assert`:
```d
import std.container.array;
import std.stdio;
void main()
{
    Array!int x;
    x.reserve(size_t.max);
}
```